packages/rs-json-schema-compatibility-validator/src/rules/rule_set.rs

```diff
@@ -0,0 +1,1378 @@
+                    allow_removal: true,
+                    allow_replacement_callback: U64_SMALLER_CALLBACK.clone(),
+                    subschema_levels_depth: None,
+                    inner: None,
+                    #[cfg(any(test, feature = "examples"))]
+                    examples: vec![
+                        (
+                            json!({}),
+                            json!({ "minLength": 1 }),
+                            Some(JsonSchemaChange::Add(AddOperation {
+                                path: "/minLength".to_string(),
+                                value: json!(1),
+                            })),
+                        )
+                            .into(),
+                        (json!({ "minLength": 1 }), json!({}), None).into(),
+                        (
+                            json!({ "minLength": 1 }),
+                            json!({ "minLength": 2 }),
+                            Some(JsonSchemaChange::Replace(ReplaceOperation {
+                                path: "/minLength".to_string(),
+                                value: json!(2),
+                            })),
+                        )
+                            .into(),
+                        (json!({ "minLength": 2 }), json!({ "minLength": 1 }), None).into(),
+                    ],
+                },
+            ),
+            (
+                "pattern",
+                CompatibilityRules {
+                    allow_addition: false,
+                    allow_removal: true,
+                    allow_replacement_callback: FALSE_CALLBACK.clone(),
+                    subschema_levels_depth: None,
+                    inner: None,
+                    #[cfg(any(test, feature = "examples"))]
+                    examples: vec![
+                        (
+                            json!({}),
+                            json!({ "pattern": "[a-z]" }),
+                            Some(JsonSchemaChange::Add(AddOperation {
+                                path: "/pattern".to_string(),
+                                value: json!("[a-z]"),
+                            })),
+                        )
+                            .into(),
+                        (json!({ "pattern": "[a-z]" }), json!({}), None).into(),
+                        (
+                            json!({ "pattern": "[a-z]" }),
+                            json!({ "pattern": "[0-9]" }),
+                            Some(JsonSchemaChange::Replace(ReplaceOperation {
+                                path: "/pattern".to_string(),
+                                value: json!("[0-9]"),
+                            })),
+                        )
+                            .into(),
+                    ],
+                },
+            ),
+            (
+                "maxItems",
+                CompatibilityRules {
+                    allow_addition: false,
+                    allow_removal: true,
```

Comment:
I don't understand why we would allow the removal here.

Reply:
If we don't need this constraint anymore we can remove it and the existing documents will be still valid. The main point of "compatibility" is that existing data must be valid with a new schema.